cmd/mayactl/app/command/volume_info.go

```diff
@@ -0,0 +1,177 @@
+package command
+
+import (
+	"errors"
+	"fmt"
+	"html/template"
+	"os"
+	"strconv"
+	"strings"
+
+	client "github.com/openebs/maya/pkg/client/jiva"
+	"github.com/openebs/maya/pkg/util"
+	"github.com/openebs/maya/types/v1"
+	"github.com/spf13/cobra"
+)
+
+var (
+	volumeInfoCommandHelpText = `
+	    Usage: mayactl volume info --volname <vol>
+
+        This command fetches the information and status of the various
+	    aspects of the Volume such as ISCSI, Controller and Replica.
+        `
+)
+
+// PortalInfo keep info about the ISCSI Target Portal.
+type PortalInfo struct {
+	IQN        string
+	VolumeName string
+	Portal     string
+	Size       string
+	Status     string
+}
+
+// ReplicaInfo keep info about the replicas.
+type ReplicaInfo struct {
+	IP         string
+	AccessMode string
+	Status     string
+}
+
+// CmdVolumeInfoOptions is used to store the value of flags used in the cli
+type CmdVolumeInfoOptions struct {
+	volName string
+}
+
+// NewCmdVolumeInfo shows info of OpenEBS Volume
+func NewCmdVolumeInfo() *cobra.Command {
+	options := CmdVolumeInfoOptions{}
+	cmd := &cobra.Command{
+		Use:     "info",
+		Short:   "Displays the info of Volume",
+		Long:    volumeInfoCommandHelpText,
+		Example: `mayactl volume info --volname <vol>`,
+		Run: func(cmd *cobra.Command, args []string) {
+			util.CheckErr(options.Validate(cmd), util.Fatal)
+			util.CheckErr(options.RunVolumeInfo(cmd), util.Fatal)
+		},
+	}
+	cmd.Flags().StringVarP(&options.volName, "volname", "n", options.volName,
+		"unique volume name.")
+	return cmd
+}
+
+// Validate verifies the command whether volName is passed or not.
+func (c *CmdVolumeInfoOptions) Validate(cmd *cobra.Command) error {
+	if c.volName == "" {
+		return errors.New("--volname is missing. Please try running [mayactl volume list] to see list of volumes.")
+	}
+	return nil
+}
+
+// TODO : Add more volume information
+// RunVolumeInfo runs info command and make call to DisplayVolumeInfo
+func (c *CmdVolumeInfoOptions) RunVolumeInfo(cmd *cobra.Command) error {
+	annotation := &Annotations{}
+	// GetVolumeAnnotation is called to get the volume controller's info such as
+	// controller's IP, status, iqn, replica IPs etc.
+	err := annotation.GetVolAnnotations(c.volName)
+	if err != nil {
+		return nil
+	}
+	if annotation == nil {
+		fmt.Println("annotation is nil")
```

Comment:
if annotation == nil, the above line 79 will it panic? Also change the message "annotation is nil" to something more user friendly like - "Unable to fetch volume details. Error: .." and print the error details.

Reply:
@kmova This is not an error, this happens when volumes have not came yet.So it will come of the function.